pkg/registry/job/strategy.go

```diff
@@ -163,6 +165,23 @@ func JobToSelectableFields(job *batch.Job) fields.Set {
 	return generic.MergeFieldsSets(objectMetaFieldsSet, specificFieldsSet)
 }
 
+func CheckJob(r *registry.Store, ctx api.Context, err error, obj runtime.Object) error {
```

Comment:
I don't see a point in having user-friendly message just for certain resources. If anything, I'd rather see a general already exists error that can be printed to all resources without any additional internal object condition check.